steal-stache.js

```diff
@@ -5,21 +5,31 @@ var loader = require("@loader");
 var addImportSpecifiers = require("steal-config-utils/import-specifiers").addImportSpecifiers;
 
 function template(imports, intermediate, filename){
+	var tagImportNames = JSON.stringify(imports.slice(7));
 	imports = JSON.stringify(imports);
 	intermediate = JSON.stringify(intermediate);
 
-	return "define("+imports+",function(module, assign, stache, mustacheCore){ \n" +
+	return "define("+imports+",function(module, assign, stache, mustacheCore, Scope, canViewImport, canStacheBindings){ \n" +
 		(filename ?
 			"\tvar renderer = stache(" + JSON.stringify(filename) + ", " + intermediate + ");\n" :
 			"\tvar renderer = stache(" + intermediate + ");\n"
 		) +
+		"\tvar tagImports = Array.prototype.slice.call(arguments, 7);\n" +
 		"\treturn function(scope, options, nodeList){\n" +
 		"\t\tvar moduleOptions = assign({}, options);\n" +
-		"\t\tif(moduleOptions.helpers) {\n" +
-		"\t\t\tmoduleOptions.helpers = assign({ module: module }, moduleOptions.helpers);\n" +
-		"\t\t} else {\n" +
-		"\t\t\tmoduleOptions.module = module;\n" +
+		"\t\tvar tagImportMap = " + tagImportNames + ".reduce(function(map, name, index) {\n" +
+		"\t\t\tmap[name] = tagImports[index];\n" +
+		"\t\t\treturn map;\n" +
+		"\t\t}, {});\n" +
+		"\n"+
+		"\t\tif (!(scope instanceof Scope)) { scope = new Scope(scope); }\n" +
+		"\t\tvar variableScope = scope.getScope(function(s) { return s._meta.variable === true });\n" +
```

Comment:
Should I change this to only check the "leaf" scope to see if it's a LetContext? This currently scans for the closest LetContext which may not be what we want.

Reply:
Maybe we should always be adding a LetContext :/